src/format/parser/ImportNode.ts

```diff
@@ -262,14 +268,18 @@ export default class ImportNode extends Statement {
   }
 }
 
-function isNameUsed(name: NameBinding | string | undefined, usage: NameUsage) {
+function isNameUsed(
+  name: NameBinding | string | undefined,
+  usage: NameUsage,
+  keepUnused: (name: string) => boolean,
+) {
   if (!name) return false;
   const { unusedNames, usedNames } = usage;
   const n = typeof name === 'string' ? name : name.aliasName ?? name.propertyName;
   if (!n) return false;
   // `unusedNames` (from TS compiler) gives more accurate results
   // than `usedNames` (from manual parsing).
-  if (unusedNames) return !unusedNames.has(n);
+  if (unusedNames) return !unusedNames.has(n) || keepUnused(n);
```

Comment:
`keepUnused(n)` should  be standalone and before this `if`.
```suggestion
  if (keepUnused(n)) return true;
  if (unusedNames) return !unusedNames.has(n);
```

Reply:
I guess if it's supposed to be kept it doesn't matter if it's unused or not, good call.